Xamarin.Forms.Build.Tasks/CompiledConverters/RDSourceTypeConverter.cs

```diff
@@ -22,7 +23,23 @@ public IEnumerable<Instruction> ConvertFromString(string value, ILContext contex
 
 			var rdNode = node.Parent as IElementNode;
 
-			var rootTargetPath = XamlCTask.GetPathForType(module, ((ILRootNode)rootNode).TypeReference);
+			var rootTargetPath = XamlCTask.GetPathForType(currentModule, ((ILRootNode)rootNode).TypeReference);
+
+			var module = currentModule;
+			string asmName = null;
+			if (value.Contains(";assembly="))
+			{
+				var parts = value.Split(new[] { ";assembly=" }, StringSplitOptions.RemoveEmptyEntries);
+				value = parts[0];
+				asmName = parts[1];
+				if (currentModule.Assembly.Name.Name != asmName)
+				{
+					var ar = currentModule.AssemblyReferences.FirstOrDefault(ar => ar.Name == asmName);
```

Comment:
Should we check if the assembly exists ? And throw a more meaningful  error if it doesn't  , let's say i set something like:

```
SharedResourceDictionary.xaml;assembly=FooWithANNamingError",
````

Reply:
you get ResourceMissing. We could have a specialised message, but not in a -sr (as it involves translation)

Reply:
I tried to add a comment here but must have missed to submit it.

Wouldn't it be safer to split the string on '=' and parse the left and right parts independently? The code right now assumes that 'assembly=' is the last parameter. It will also crash if 'assembly=' is used, which to the best of my knowledge is valid and should just be ignored.